ouroboros-network/src/Ouroboros/Network/Block.hs

```diff
@@ -108,27 +116,37 @@ castHash (BlockHash b) = BlockHash b
   Point on a chain
 -------------------------------------------------------------------------------}
 
--- | A point on the chain is identified by its 'Slot' and 'HeaderHash'.
---
--- The 'Slot' tells us where to look and the 'HeaderHash' either simply serves
--- as a check, or in some contexts it disambiguates blocks from different forks
--- that were in the same slot.
---
-data Point block = Point {
-       pointSlot :: SlotNo,
-       pointHash :: ChainHash block
-     }
-   deriving (Eq, Ord, Show)
+-- | A block is identified by a slot and a hash.
+data TBlockPoint slot hash = TBlockPoint
+  { pointSlot :: slot
+  , pointHash :: hash
+  }
+  deriving (Eq, Ord, Show)
+
+type BlockPoint block = TBlockPoint SlotNo (HeaderHash block)
+
+-- | `BlockPoint` with an origin point.
+data TPoint t where
```

Comment:
Hmm, why not call this type just `Point`?  I also think that using `block` instead of `t` is better, for the consistency sake.  A comment can explain that this can be a block or header, and a comment could explain that `Origin :: TPoing block` corresponds to `Genesis :: Chain block`.

Reply:
Would be better as `Point` but I used a new name so that I wouldn't have to make as many name changes in downstream modules.

Reply:
I see, we can claim that `T` stands for `ThePoint`.

Reply:
I think we all agree that distinguishing the genesis point from other block points is a good thing, lets do it fully, even if it means more downstream changes. Fewer names in the API is a good thing.